### ISSUE TITLE
Error Parsing Script for DDLDatabase when column name is called "index"

### ISSUE THREAD

**Issue by Zidane** [jOOQ/jOOQ#7651]
### Expected behavior and actual behavior:
I have a script I used with codegen to generate Java classes via DDLDatabase. After updating from 3.6 to 3.11, I now get the following parser exception:

```
SEVERE: An exception occurred while parsing script source : src/main/resources/db/axs/database.sql. Please report this error to https://github.com/jOOQ/jOOQ/issues/new
org.jooq.impl.ParserException: Token '(' expected: [19:33] ...lt (0) not null,
  index            int          [*]default (0) not null,
  is_hidden        bit          default (0) not null,
  ...
        at org.jooq.impl.ParserContext.expected(ParserImpl.java:8443)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:8010)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:8005)
        at org.jooq.impl.ParserImpl.parseIndexSpecification(ParserImpl.java:2632)
        at org.jooq.impl.ParserImpl.parseCreateTable(ParserImpl.java:2286)
        at org.jooq.impl.ParserImpl.parseCreate(ParserImpl.java:1835)
        at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:673)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:444)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:431)
        at org.jooq.meta.extensions.ddl.DDLDatabase.load(DDLDatabase.java:199)
        at org.jooq.meta.extensions.ddl.DDLDatabase.load(DDLDatabase.java:178)
        at org.jooq.meta.extensions.ddl.DDLDatabase.create0(DDLDatabase.java:138)
        at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:231)
        at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:220)
        at org.jooq.meta.AbstractDatabase.getDialect(AbstractDatabase.java:202)
        at org.jooq.codegen.AbstractGenerator.logDatabaseParameters(AbstractGenerator.java:155)
        at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:297)
        at org.jooq.codegen.GenerationTool.run(GenerationTool.java:750)
        at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:220)
        at org.jooq.codegen.GenerationTool.main(GenerationTool.java:192)
```

Script:

```
create table if not exists axs
(
  rec_no           int          auto_increment,
  created          timestamp    default current_timestamp not null,
  creator          binary(16)   not null,
  id               varchar(255) primary key,
  name             varchar(255) not null
);

create table if not exists axs_store_item
(
  rec_no           int          primary key auto_increment,
  created          timestamp    default current_timestamp not null,
  axs              varchar(255) not null,
  seller           binary(16)   not null,
  item_type        varchar(255) not null,
  initial_quantity int          default 1 not null,
  price            decimal      default 0 not null,
  index            int          default 0 not null,
  is_hidden        bit          default 0 not null,
  foreign key (axs) references axs(id) on update cascade on delete cascade
);

create table if not exists axs_store_item_data
(
  rec_no           int          auto_increment,
  store_item       int          primary key,
  data             varbinary    not null,
  foreign key (store_item) references axs_store_item(rec_no) on update cascade on delete cascade
);

create table if not exists axs_list_item
(
  rec_no           int          auto_increment,
  created          timestamp    default current_timestamp not null,
  store_item       int          primary key auto_increment,
  is_hidden        bit          default 0 not null,
  quantity         int          default 1 not null,
  foreign key (store_item) references axs_store_item(rec_no) on update cascade on delete cascade
);

create table if not exists axs_transaction
(
  rec_no           int          primary key auto_increment,
  created          timestamp    default current_timestamp not null,
  list_item        int          not null,
  buyer            binary(16)   not null,
  quantity         int          default 1 not null,
  foreign key (list_item) references axs_store_item(rec_no) on update cascade on delete cascade
);
```

### Versions:

- jOOQ: 3.11
- Java: 1.8
- Database (include vendor): h2
- OS: Windows 8.1
- JDBC Driver (include name if inofficial driver): h2

**Comment by lukaseder** [comment]
Thank you very much for your report. I can reproduce this. The problem is the column name `index`, which in MySQL could be used to describe an inline index definition:

```sql
CREATE TABLE t (
  a INT,
  index i (a)
);
```

We'll have to disambiguate those index definitions a bit better by looking ahead a few tokens.

In the meantime, a workaround would be to quote the `index` column. This should work:

```sql
create table if not exists axs_store_item
(
  rec_no           int          primary key auto_increment,
  created          timestamp    default current_timestamp not null,
  axs              varchar(255) not null,
  seller           binary(16)   not null,
  item_type        varchar(255) not null,
  initial_quantity int          default 1 not null,
  price            decimal      default 0 not null,
  "index"          int          default 0 not null,
  is_hidden        bit          default 0 not null,
  foreign key (axs) references axs(id) on update cascade on delete cascade
);
```

**Comment by Zidane** [comment]
@lukaseder 

Workaround confirmed working. I should have thought of that as you need to do this anyways to have codegen generate Fields that use lowercase namesand tables with lowercase names too. Perhaps both of these quirks can be solved at the same time?

**Comment by lukaseder** [comment]
> Perhaps both of these quirks can be solved at the same time?

No, they're unrelated. This `index` name thing can be solved (and backported) much more easily than the case-sensitivity thing, which is a new feature.

**Comment by lukaseder** [comment]
Fixed for 3.12. Backport scheduled for 3.11.3: #7657
